### ISSUE TITLE
Many users are not aware of the filtering keywords available for the search

### ISSUE THREAD

**Issue by maximelehericy** [nextcloud/server#33102]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Is your feature request related to a problem? Please describe.**
I am always frustrated when I'd like to filter a search of a specific item, but cannot remember the right keyword (e.g. in:talk-message), and that I have to wait the end of the search (sometimes long... :/) to access the three dot menu enabling me to narrow the search.

**Describe the solution you'd like**
Once the search finished, there is a three dot menu net to the search field, to help narrow the search.
Why doesn't this menu available right from the beginning ? Next to the search field ?
That would bring several improvement in my opinion:
1. ability to filter a search right from the beginining
2. make users conscious/more aware that there are keywords that they can use to narrow don their search.

**Describe alternatives you've considered**
Write the keywords in light grey in the search field when it is empty.
![image](https://user-images.githubusercontent.com/87175513/177160947-9a8c87bf-d945-42c2-ae0a-7a9bfc89a812.png)
(replace "search in Files..." by "filter with in:files, etc..."

**Additional context**
![image](https://user-images.githubusercontent.com/87175513/177160203-2afb53f0-9e89-4038-bd98-d565ed3c0f30.png)

**Comment by XueSheng-GIT** [comment]
I would agree that search isn't intuitive enough if you need to filter the results. Additionally to the aspects mentioned by @maximelehericy there also seems to be a search syntax which I couldn't find a documentation for (like using "+", "-" or "*", which on the other side does limit the search result to fulltextsearch... which isn't obvious too).

**Comment by PVince81** [comment]
@jancborchardt design feedback from Maxime

**Comment by maximelehericy** [comment]
smart :)

**Comment by ishaan730** [comment]
Hi @jancborchardt,

Another potential design option to consider would be to group the filter chips together directly below the search bar. This approach would work well for our current state, particularly if users frequently need to filter search results.

Incorporating this design pattern could have several benefits for the user experience. Firstly, it would provide users with a clear and accessible way to apply filters and refine their search results. Secondly, by placing the filters in close proximity to the search bar, users can easily understand the relationship between the two elements, reducing the risk of confusion or frustration. Finally, the use of filter chips can provide users with a visual representation of the filters they have applied, making it easier for them to understand their search results and refine their filters as needed.

Additionally, this design choice has the potential to significantly improve the usability and effectiveness of our team chat app's search feature.

**Here's a quick mockup for reference**

<img width="392" alt="Screenshot 2023-04-27 at 12 56 58 AM" src="https://user-images.githubusercontent.com/49994669/234682083-ba3e30e9-19e3-47cf-af1b-40312ccd46d6.png">

Or, 

Another design option could be, Adding a filter icon button right inside the search bar. This will provide users with a convenient and intuitive way to quickly filter search results based on specific criteria, improving their ability to find the information they need.

By placing the filter icon button inside the search bar, we can help users to easily locate and access this functionality. This can also help to reduce clutter on the interface, as users do not need to navigate to a separate button or section of the app to access the filters.

In addition, using an icon for the filter button can help to conserve valuable screen real estate, as icons take up less space than text. This can be particularly important in a chat app where space is at a premium and users may need to quickly navigate between conversations.

**Here's quick ref. mockup** 

<img width="456" alt="Screenshot 2023-04-27 at 1 05 49 AM" src="https://user-images.githubusercontent.com/49994669/234683977-2fce0510-13c5-4957-bee8-297f4db48f53.png">

Please let me know your thoughts on this. Thank you!

**Comment by joshtrichards** [comment]
Closing as completed - this was addressed in >=v28 with the new search design.
